sequencer/src/state.rs

```diff
@@ -1202,4 +1271,92 @@ mod test {
     }
+
+    #[test]
+    fn test_validation_max_block_size() {
+        setup_logging();
+        setup_backtrace();
+
+        const MAX_BLOCK_SIZE: usize = 10;
+        let payload = [0; 2 * MAX_BLOCK_SIZE];
+        let vid_common = vid_scheme(1).disperse(payload).unwrap().common;
+
+        let state = ValidatedState::default();
+        let instance = NodeState::mock().with_chain_config(ChainConfig {
+            max_block_size: MAX_BLOCK_SIZE as u64,
+            base_fee: 0.into(),
+            ..Default::default()
+        });
+        let parent = Leaf::genesis(&instance);
+        let header = parent.get_block_header();
+
+        // Validation fails because the proposed block exceeds the maximum block size.
+        let err = validate_proposal(&state, instance.chain_config, &parent, header, &vid_common)
+            .unwrap_err();
+        tracing::info!(%err, "task failed successfully");
+    }
+
+    #[test]
+    fn test_validation_base_fee() {
+        setup_logging();
+        setup_backtrace();
+
+        let max_block_size = 10;
+        let payload = [0; 1];
+        let vid_common = vid_scheme(1).disperse(payload).unwrap().common;
+
+        let state = ValidatedState::default();
+        let instance = NodeState::mock().with_chain_config(ChainConfig {
+            base_fee: 1000.into(), // High base fee
+            max_block_size,
+            ..Default::default()
+        });
+        let parent = Leaf::genesis(&instance);
+        let header = parent.get_block_header();
+
+        // Validation fails because the genesis fee (0) is too low.
+        let err = validate_proposal(&state, instance.chain_config, &parent, header, &vid_common)
+            .unwrap_err();
+        tracing::info!(%err, "task failed successfully");
+    }
+
+    #[test]
+    fn test_burn_fee() {
+        setup_logging();
+        setup_backtrace();
+
+        let src = FeeAccount::generated_from_seed_indexed([0; 32], 0).0;
+        let dst = FeeAccount::generated_from_seed_indexed([0; 32], 1).0;
+        let amt = FeeAmount::from(1);
+
+        let fee_info = FeeInfo::new(src, amt);
+
+        let new_state = || {
+            let mut state = ValidatedState::default();
+            state.prefund_account(src, amt);
+            state
+        };
+
+        tracing::info!("test successful burn");
+        let mut state = new_state();
+        state.burn_fee(fee_info, dst).unwrap();
+        assert_eq!(state.balance(src), Some(0.into()));
+        assert_eq!(state.balance(dst), Some(amt));
+
+        tracing::info!("test insufficient balance");
+        state.burn_fee(fee_info, dst).unwrap_err();
+        assert_eq!(state.balance(src), Some(0.into()));
+        assert_eq!(state.balance(dst), Some(amt));
+
+        tracing::info!("test src not in memory");
+        let mut state = new_state();
+        state.fee_merkle_tree.forget(src).expect_ok().unwrap();
+        state.burn_fee(fee_info, dst).unwrap_err();
```

Comment:
Here as well I think if we have tests that are trying to hit a specific type of validation error we should make sure that it did hit the right type of validation error. Otherwise it's easy for tests to no longer fail for the right reasons as the code base evolves. Conversely I think it's preferable to have well defined error types instead of using `anyhow` so I think we should move away from using ad-hoc anyhow errors for all our validation code. If there's potential consensus on this let's open an issue and redo all the validation errors in a separate PR.

Reply:
Yeah, I think let's do this in a separate PR

Reply:
https://github.com/EspressoSystems/espresso-sequencer/issues/1463